ballerina/auth_desugar.bal

```diff
@@ -54,89 +54,122 @@ public isolated function authenticateResource(Service serviceRef, string methodN
 
 isolated function tryAuthenticate(ListenerAuthConfig[] authConfig, string header) returns Unauthorized|Forbidden? {
+    string scheme = extractScheme(header);
+    Unauthorized|Forbidden? authResult = <Unauthorized>{};
     foreach ListenerAuthConfig config in authConfig {
-        if config is FileUserStoreConfigWithScopes {
-            ListenerFileUserStoreBasicAuthHandler handler;
-            lock {
-                string key = config.fileUserStoreConfig.toString();
-                if authHandlers.hasKey(key) {
-                    handler = <ListenerFileUserStoreBasicAuthHandler> authHandlers.get(key);
-                } else {
-                    handler = new(config.fileUserStoreConfig.cloneReadOnly());
-                    authHandlers[key] = handler;
-                }
-            }
-            auth:UserDetails|Unauthorized authn = handler.authenticate(header);
-            string|string[]? scopes = config?.scopes;
-            if authn is auth:UserDetails {
-                if scopes is string|string[] {
-                    Forbidden? authz = handler.authorize(authn, scopes);
-                    return authz;
-                }
-                return;
-            }
-        } else if config is LdapUserStoreConfigWithScopes {
-            ListenerLdapUserStoreBasicAuthHandler handler;
-            lock {
-                string key = config.ldapUserStoreConfig.toString();
-                if authHandlers.hasKey(key) {
-                    handler = <ListenerLdapUserStoreBasicAuthHandler> authHandlers.get(key);
-                } else {
-                    handler = new(config.ldapUserStoreConfig.cloneReadOnly());
-                    authHandlers[key] = handler;
-                }
-            }
-            auth:UserDetails|Unauthorized authn = handler->authenticate(header);
-            string|string[]? scopes = config?.scopes;
-            if authn is auth:UserDetails {
-                if scopes is string|string[] {
-                    Forbidden? authz = handler->authorize(authn, scopes);
-                    return authz;
-                }
-                return;
-            }
-        } else if config is JwtValidatorConfigWithScopes {
-            ListenerJwtAuthHandler handler;
-            lock {
-                string key = config.jwtValidatorConfig.toString();
-                if authHandlers.hasKey(key) {
-                    handler = <ListenerJwtAuthHandler> authHandlers.get(key);
-                } else {
-                    handler = new(config.jwtValidatorConfig.cloneReadOnly());
-                    authHandlers[key] = handler;
-                }
+        if scheme is AUTH_SCHEME_BASIC {
+            if config is FileUserStoreConfigWithScopes {
+                authResult = authenticateWithFileUserStore(config, header);
+            } else if config is LdapUserStoreConfigWithScopes {
+                authResult = authenticateWithLdapUserStoreConfig(config, header);
             }
```

Comment:
We can add a debug log for else case to understand the flow when solving issues

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-http/pull/794/commits/5ab10c20d35dd80a266c22fb5fc064d92f14f7d6